src/plugins/camera_tracking/CameraTracking.cc

```diff
@@ -0,0 +1,509 @@
+               msgs::Boolean &_res);
+
+  /// \brief Callback for a follow offset request
+  /// \param[in] _msg Request message to set the camera's follow offset.
+  /// \param[in] _res Response data
+  /// \return True if the request is received
+  public: bool OnFollowOffset(const msgs::Vector3d &_msg,
+               msgs::Boolean &_res);
+
+  /// \brief Callback when a move to animation is complete
+  private: void OnMoveToComplete();
+
+  /// \brief Callback when a move to  pose animation is complete
+  private: void OnMoveToPoseComplete();
+
+  /// \brief Process key releases
+  /// \param[in] _e Key release event
+  public: void HandleKeyRelease(events::KeyReleaseOnScene *_e);
+
+  /// \brief Protects variable changed through services.
+  public: std::mutex mutex;
+
+  //// \brief Pointer to the rendering scene
+  public: rendering::ScenePtr scene = nullptr;
+
+  /// \brief Target to follow
+  public: std::string followTarget;
+
+  /// \brief Wait for follow target
+  public: bool followTargetWait = false;
+
+  /// \brief Offset of camera from target being followed
+  public: math::Vector3d followOffset = math::Vector3d(-5, 0, 3);
+
+  /// \brief Flag to indicate the follow offset needs to be updated
+  public: bool followOffsetDirty = false;
+
+  /// \brief Flag to indicate the follow offset has been updated
+  public: bool newFollowOffset = true;
+
+  /// \brief Follow P gain
+  public: double followPGain = 0.01;
+
+  /// \brief True follow the target at an offset that is in world frame,
+  /// false to follow in target's local frame
+  public: bool followWorldFrame = false;
+
+  /// \brief Last move to animation time
+  public: std::chrono::time_point<std::chrono::system_clock> prevMoveToTime;
+
+  /// \brief User camera
+  public: rendering::CameraPtr camera{nullptr};
+
+  /// \brief Target to move the user camera to
+  public: std::string moveToTarget;
+
+  /// \brief Helper object to move user camera
+  public: ignition::rendering::MoveToHelper moveToHelper;
+
+  /// \brief Transport node
+  public: transport::Node node;
+
+  /// \brief Move to service
+  public: std::string moveToService;
+
+  /// \brief The pose set from the move to pose service.
+  public: std::optional<math::Pose3d> moveToPoseValue;
+
+  /// \brief Follow service
+  public: std::string followService;
+
+  /// \brief Follow offset service
+  public: std::string followOffsetService;
+
+  /// \brief Camera pose topic
+  public: std::string cameraPoseTopic;
+
+  /// \brief Move to pose service
+  public: std::string moveToPoseService;
+
+  /// \brief Camera pose publisher
+  public: transport::Node::Publisher cameraPosePub;
+
+  /// \brief Timer to keep publishing camera poses.
+  public: QTimer *timer{nullptr};
+};
+
+using namespace ignition;
+using namespace gui;
+using namespace plugins;
+
+/////////////////////////////////////////////////
+void CameraTrackingPrivate::Initialize()
+{
+  // Attach to the first camera we find
```

Comment:
We have to sort this out to make sure we're not attaching to a plugin, see https://github.com/ignitionrobotics/ign-gui/pull/231#discussion_r658353941

Reply:
Let's get this in as is for now and handle choosing the camera in a future PR, added an item here: https://github.com/ignitionrobotics/ign-gui/issues/137#issuecomment-875124456